common/lib/xmodule/xmodule/course_module.py

```diff
@@ -1601,3 +1602,72 @@ def can_toggle_course_pacing(self):
           bool: False if the course has already started, True otherwise.
         """
         return datetime.now(UTC()) <= self.start
+
+
+class CourseSummaryDescriptor(object):
+    """
+    A lightweight course summary class, which parses split/mongo course information without loading
+    the course. It is used at cms for listing courses to global staff user.
+    """
+
+    def __init__(self, store, course_structure, course_locator):
+        """
+        Initialize and parse the `course_structure`
+        """
+        self.default_display_name = "Empty"
+        self.display_coursenumber = None
+        self.display_organization = None
+        self.course_info_fields = ['display_name', 'display_coursenumber', 'display_organization']
+
+        self.course_structure = course_structure
+
+        self.id = course_locator  # pylint: disable=invalid-name
+        self.location = course_locator.make_usage_key('course', 'course')
+
+        # Identify the store and extract course summary
+        if store == ModuleStoreEnum.Type.mongo:
+            self.extract_course_summary_for_mongo()
+        elif store == ModuleStoreEnum.Type.split:
+            self.extract_course_summary_split()
```

Comment:
Rather than making this object have knowledge of multiple modulestores, have the modulestores pass in the data into this object. That will keep the code that pertains to how data is stored centralized to the code responsible for it (the modulestore), rather than spread across multiple places.